05-JWT-Basics/preferred/controllers/main.js

```diff
@@ -0,0 +1,30 @@
+const jwt = require('jsonwebtoken');
+
+//user inserts name, password making POST request
+const logon = async (req, res) => {
+  const { name, password } = req.body;
+  if (!name || !password || name === '' || password === '') {
+    return res
+      .status(400)
+      .json({
+        message:
+          'You should provide BOTH name AND password to get access to Secret Data',
+      });
+  }
+  const token = jwt.sign({ name, password }, process.env.JWT_SECRET_KEY, {
```

Comment:
Not a super big deal here since this is just a practice app; but just want to note that for production apps, we never want to put the password in the JWT.

This is because the JWT is encoded to be not-human-readable, _**but**_ it is not cryptographically encrypted or hashed or anything, so it's very easy for anyone to stick it in a JWT decoder and read the values. That means if a hacker has a way to read requests that are happening on your website, they can get people's passwords this way. You can try it out by taking the token that this code creates, and sticking it into a website like this: [https://jwt.io/#debugger-io](https://jwt.io/#debugger-io)

Reply:
I think I saw some comments you have in the `starter` directory about this so you probably already know, but with security stuff it's always good to be super sure :sweat\_smile:

Reply:
@akosasante omg! Thank you so much for your comments ! I totally forgot about it! I definitely need to watch again all the lessons , I don’t feel fluent about what I’m doing  …  please, point me where I am wrong ,  because if there is any chance to make a mistake , I’m gonna make it 😃 ! Love your feedback!